src/canvas/SelectableCanvas.ts

```diff
@@ -680,31 +686,49 @@ export class SelectableCanvas<
           target.cacheTranslationY + normalizedPointer.y * target.zoomY,
           0
         );
+      // transform tolerance according to vpt
+      // TODO: use sendVectorToPlane
+      const tolerance = new Point()
+        .scalarAdd(this.targetFindTolerance)
+        .transform(invertTransform(this.viewportTransform), true);
+      const size = tolerance.scalarMultiply(2).max(new Point(1, 1));
+      // performance optimization:
+      // we draw the hit area to the dedicated canvas instead of using `getImageData` on the target's cache canvas
+      // since `size` is transformed according to vpt the image is drawn as if transformed as well so `targetFindTolerance` can be used as the tolerance value
+      ctx.drawImage(
+        target._cacheCanvas,
+        Math.floor(targetRelativeX - tolerance.x),
+        Math.floor(targetRelativeY - tolerance.y),
+        Math.ceil(size.x),
+        Math.ceil(size.y),
+        0,
+        0,
+        Math.ceil(size.x * retina),
+        Math.ceil(size.y * retina)
+      );
 
       return isTransparent(
-        target._cacheContext,
-        Math.round(targetRelativeX),
-        Math.round(targetRelativeY),
+        ctx,
+        this.targetFindTolerance,
+        this.targetFindTolerance,
         this.targetFindTolerance
       );
     }
 
-    const ctx = this.contextCache,
-      originalColor = target.selectionBackgroundColor,
-      v = this.viewportTransform;
-
-    target.selectionBackgroundColor = '';
-
-    this.clearContext(ctx);
-
     ctx.save();
-    ctx.transform(v[0], v[1], v[2], v[3], v[4], v[5]);
+    ctx.translate(-x + this.targetFindTolerance, -y + this.targetFindTolerance);
+    ctx.transform(...this.viewportTransform);
+    const selectionBgc = target.selectionBackgroundColor;
+    target.selectionBackgroundColor = '';
     target.render(ctx);
+    target.selectionBackgroundColor = selectionBgc;
```

Comment:
to be honest i think `selectionBackgroundColor` is just a terrible feature and that it could be deleted, but we shouldn't do it here.
Is a property that pertain the control bounding box and was added to the objects because rush/inexperience

Reply:
I met it redoing controls. Managed to get it working there but I agree it is weird because it seems not part of object rendering scope, same as controls, but because it needs to be behind and object stacking I understand why you did it.